src/components/FormField/Input/TextField.tsx

```diff
@@ -0,0 +1,63 @@
+import { InputHTMLAttributes, forwardRef, useId, useRef } from "react";
+import { Icon } from "../..";
+import { IconButton, InputElement, InputWrapper, WrapperProps } from "./InputWrapper";
+import { mergeRefs } from "@/utils/mergeRefs";
+import { mockInputChangeEvent } from "./mockInputChangeEvent";
+interface TextInputProps
+  extends Omit<WrapperProps, "id" | "children">,
+    Omit<InputHTMLAttributes<HTMLInputElement>, "children" | "type" | "value"> {
+  type?: "text" | "email" | "tel" | "url";
+  loading?: boolean;
+  value?: string;
+  clear?: boolean;
+}
+
+export const TextField = forwardRef<HTMLInputElement, TextInputProps>(
+  (
+    { clear, type, disabled, label, error, id, loading, value = "", onChange, ...props },
+    ref
+  ) => {
+    const inputRef = useRef<HTMLInputElement>(null);
+    const defaultId = useId();
+    const clearInput = () => {
+      mockInputChangeEvent(inputRef.current, onChange);
+    };
+
+    return (
+      <InputWrapper
+        disabled={disabled}
+        id={id ?? defaultId}
+        label={label}
+        error={error}
+      >
+        <InputElement
+          ref={mergeRefs([inputRef, ref])}
+          type={type}
+          id={id ?? defaultId}
+          disabled={disabled}
+          value={value}
+          onChange={onChange}
+          {...props}
+        />
+
+        {clear && value.length > 0 && (
+          <IconButton
```

Comment:
when the icon button appears, the input size changes, i think it would be better if it stays the same

Reply:
May be we can add animation to make it seamless but the input width should change so as the close button does not overlap the text
Or may be we can always show the element and hide it using css using opacity: 0

Reply:
option 2, but `visibility:hidden` is better than`opacity:0` as in the second case the element is still there and the users can interact with it

Reply:
Fixed